src/cr/cube/cube_slice.py

```diff
@@ -479,6 +480,59 @@ def zscore(self, weighted=True, prune=False, hs_dims=None):
 
         return zscore
 
+    def compare_to_column(self, column=0, weighted=True):
+        """Test statistic and p-value for t-tests of proportions compared
+        to a column
+
+        Returns TtestResult, a pair of arrays of the same size as the current
+        slice: *statistic*, the t-statistic that proportions are different from
+        column *column*; and pvalue, the two-tailed p-value associated with that
+        test statistic (df = table N-2).
+        """
+
+        counts = self.as_array(weighted=weighted)
+        props = self.proportions(axis=0)
+        diff = props[:, [column]] - props
+        margin = self.margin(axis=0, weighted=weighted)
+        var_props = props * (1.0 - props) / margin
+        se_diff = np.sqrt(var_props + var_props[:, [column]])
+        statistic = diff / se_diff
+        df = np.broadcast_to(margin, self.get_shape()) + margin[column] - 2
+        pvals = 2 * (1 - t.cdf(abs(statistic), df=df))
+        TtestResult = namedtuple("TTestResult", ("statistic", "pvalue"))
```

Comment:
We probably want an explicit class for this, if we want to group these values together. Better testing, documentation, etc...

Reply:
halp :) I was cribbing from scipy, and we should coördinate with karol for what we want to send to whaam which needs both these pieces to do coloring.